eng/common/scripts/SemVer.ps1

```diff
@@ -183,7 +183,7 @@ class AzureEngSemanticVersion : IComparable {
     $this.DefaultAlphaReleaseLabel = "alpha"
   }
 
-  [int] CompareTo($other)
```

Comment:
So implementing `IComparable` seems to indicate that you implement `[int] CompareTo`

That's why this is failing during compilation.

Reply:
Yeah. [Details](https://docs.microsoft.com/en-us/dotnet/api/system.icomparable.compareto?view=net-6.0)

You need to leave the type of this function `int` as it indicates the comparison result.